lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -142,6 +144,62 @@ def _get_user_partition(self, user_partition_id):
 
         raise NoSuchUserPartitionError("could not find a UserPartition with ID [{}]".format(user_partition_id))
 
+    def get_parent_unit(self):
+        """
+        Finds xblock's parent unit if it exists.
+
+        Returns:
+            xblock: Returns the parent unit xblock if it exists.
+            If no parent unit exists, returns None.
+        """
+        xblock = self
+        while xblock:
+            xblock = xblock.get_parent()
+            if xblock is None:
+                return None
+            parent = xblock.get_parent()
+            if parent is None:
+                return None
+            if parent.category == "sequential":
+                return xblock
+
+    def _has_nonsensical_access_settings(self):
+        """
+        Checks if a block's group access settings do not make sense.
+
+        By nonsensical access settings, we mean a component's access
+        settings which contradict the unit level access in that they
+        restrict access to a component in such a way that no one can
+        see it.
+        For example, if unit level access is set to "Verified", and
+        a component within the unit is set to "Audit", that setting
+        does not make sense because the component will not be
+        visible to anybody.
+        Note:  The only time this contradiction can occur is when
+        a component restricts access to the same partition but a
+        different group than the unit.
+
+        Returns:
+            bool: True if the block is a component and its access
+            settings contradict the unit level access.
+        """
+        parent_unit = self.get_parent_unit()
+        if not parent_unit:
+            return False
+
+        unit_group_access = parent_unit.group_access
+        component_group_access = self.group_access
+
+        for user_partition_id, unit_group_ids in unit_group_access.iteritems():
+            try:
+                component_group_ids = component_group_access[user_partition_id]
+                if component_group_ids and unit_group_ids and not set(unit_group_ids).issubset(set(component_group_ids)):
```

Comment:
I believe the issubset check needs to be flipped around. Consider the following (3 content groups).

1) Unit access set to Group 1 and Group2
2) Component within unit set to just Group 1

Currently a validation message is shown, but this is a perfectly legitimate case.

Plus, if we switch it around we will warn when any groups that are selected are nonsensical (as I mentioned in HipChat, you should verify with @sstack22 that this is the desired behavior). To be clear, I'm talking about this scenario:

1) Unit access is set to Group 1.
2) Component access is set to both Group 1 and Group 2. The Group 2 restriction here doesn't really make sense, though the component will be shown to everyone in Group 1.

Also update unit test coverage to cover both of these variations.